app/src/lib/trampoline/trampoline-environment.ts

```diff
@@ -21,18 +28,40 @@ export async function withTrampolineEnv<T>(
     ? getDesktopTrampolinePath()
     : getAskPassTrampolinePath()
 
-  return withTrampolineToken(async token =>
-    fn({
-      DESKTOP_PORT: await trampolineServer.getPort(),
-      DESKTOP_TRAMPOLINE_TOKEN: token,
-      GIT_ASKPASS: askPassPath,
-      DESKTOP_TRAMPOLINE_IDENTIFIER: TrampolineCommandIdentifier.AskPass,
-
-      // Env variables specific to the old askpass trampoline
-      DESKTOP_PATH: process.execPath,
-      DESKTOP_ASKPASS_SCRIPT: getAskPassScriptPath(),
-    })
-  )
+  const sshEnv = await getSSHEnvironment()
+
+  return withTrampolineToken(async token => {
+    // The code below assumes a few things in order to manage SSH key passphrases
+    // correctly:
+    // 1. `withTrampolineEnv` is only used in the functions `git` (core.ts) and
+    //    `spawnAndComplete` (spawn.ts)
+    // 2. Those two functions always thrown an error when something went wrong,
+    //    and just return a result when everything went fine.
+    //
+    // With those two premises in mind, we can safely assume that right after
+    // `fn` has been invoked, we can store the SSH key passphrase for this git
+    // operation if there was one pending to be stored.
+    try {
+      const result = await fn({
+        DESKTOP_PORT: await trampolineServer.getPort(),
+        DESKTOP_TRAMPOLINE_TOKEN: token,
+        GIT_ASKPASS: askPassPath,
+        DESKTOP_TRAMPOLINE_IDENTIFIER: TrampolineCommandIdentifier.AskPass,
+
+        ...sshEnv,
+
+        // Env variables specific to the old askpass trampoline
+        DESKTOP_PATH: process.execPath,
+        DESKTOP_ASKPASS_SCRIPT: getAskPassScriptPath(),
+      })
+
+      await storePendingSSHKeyPassphrase(token)
```

Comment:
If the app got to this point, it meant the git operation didn't fail, and therefore we can safely store any pending SSH key passphrase associated to it 😌